packages/type-utils/tests/isTypeReadonly.test.ts

```diff
@@ -82,6 +82,12 @@ describe('isTypeReadonly', () => {
             ['type Test = Readonly<ReadonlySet<string>>;'],
             ['type Test = Readonly<ReadonlyMap<string, string>>;'],
           ])('handles fully readonly sets and maps', runTests);
+
+          // Private Identifier.
+          // Note: It can't be accessed from outside of class thus exempt from the checks.
+          it.each([
+            ['class Foo { #privateField = "foo"; #privateMember() {}; }'],
```

Comment:
Missing the test case of a `readonly`: https://github.com/typescript-eslint/typescript-eslint/issues/4068#issuecomment-1019594138